addons/missionTesting/functions/fnc_runWeaponsTest.sqf

```diff
@@ -0,0 +1,158 @@
+            private _weaponAttachments = _unit weaponAccessories _weaponClassname;
+            private _scope = _weaponAttachments select 2;
+            if (_scope != "") then {
+                private _scopeConfig = configFile >> "CfgWeapons" >> _scope;
+                private _minZoom = 999; //FOV, so smaller is more zoomed in
+                {
+                    if (isNumber (_x >> "opticsZoomMin")) then {_minZoom = _minZoom min (getNumber (_x >> "opticsZoomMin"));};
+                    if (isText (_x >> "opticsZoomMin")) then {_minZoom = _minZoom min (call compile getText (_x >> "opticsZoomMin"));};
+                    nil
+                } count configProperties [_scopeConfig >> "ItemInfo" >> "OpticsModes"];
+                private _scopeImage = getText (_scopeConfig >> "picture");
+                _muzzleText = _muzzleText + format ["(<img image='%1' width='20' height='20'/>:%2x) ", _scopeImage, ((floor (2.5 / _minZoom))/10)];
+            };
+        };
+        private _hasMags = false;
+        {
+            private _magazineClassname = _x;
+            private _count = 0;
+            {
+                if ((_x select 0) == _magazineClassname) then {
+                    _hasMags = true;
+                    _usedAmmo pushBack _magazineClassname;
+                    _count = _count + (_x select 1);
+                };
+            } forEach (magazinesAmmoFull _unit);
+            if (_count > 0) then {
+                private _magPic = getText (configFile >> "CfgMagazines" >> _magazineClassname >> "picture");
+                _muzzleText = _muzzleText + format ["<img image='%1' width='16' height='16'/>:%2 ", _magPic, _count];
+            };
+        } forEach _mags;
+        if (_weaponClassname == "throw") then {
+            if (_hasMags) then {
+                _unitText = _unitText + _muzzleText;
+            };
+        } else {
+            _unitText = _unitText + _muzzleText + "]<br/>";
+        };
+    } forEach _muzzles;
+    if (_weaponClassname == "throw") then {
+        _unitText = _unitText + "]<br/>";
+    };
+};
+
+{
+    private _unit = _x;
+    private _classname = typeOf _unit;
+
+    if (!(_classname in _handledClasses)) then {
+        _handledClasses pushBack _classname;
+
+        private _displayName = getText (configfile >> "CfgVehicles" >> _classname >> "displayName");
+        private _xIcon = gettext (configfile >> "CfgVehicles" >> _classname >> "icon");
+        private _image = gettext (configfile >> "CfgVehicleIcons" >> _xIcon);
+        private _color = "";
+        private _sideArray = [];
+
+        switch (side _x) do {
+        case (west): {
+                _color = "#0088EE"; //0,0.45,0.9,1
+            };
+        case (east): {
+                _color = "#DD0000"; //0,0.45,0.9,1
+            };
+        case (resistance): {
+                _color = "#00DD00"; //0,0.45,0.9,1
+            };
+            default {
+                _color = "#880099"; //0,0.45,0.9,1
```

Comment:
This doesn't reflect colour comment

Reply:
This was whole sale copied from Pabo's weapons test elsewhere in POTATO. Have removed it here. 

Does remind me though. If this is adopted then it negates the need for the script used else where.... possibly.